DQM/L1TMonitor/src/L1TStage2Shower.cc

```diff
@@ -0,0 +1,183 @@
+    int ring = detId.ring();
+    int chamber = detId.chamber();
+    int sr = histIndexCSC.at({station, ring});
+    if (endcap == -1)
+      sr = 17 - sr;
+    float evt_wgt = (station > 1 && ring == 1) ? 0.5 : 1.0;
+
+    auto cscShower = (*element).second.first;
+    auto cscShowerEnd = (*element).second.second;
+    for (; cscShower != cscShowerEnd; ++cscShower) {
+      if (station > 1 && (ring % 2) == 1) {
+        if (cscShower->isLooseInTime())
+          cscShowerOccupancyLoose->Fill(chamber * 2, sr, evt_wgt);
+        if (cscShower->isNominalInTime())
+          cscShowerOccupancyNom->Fill(chamber * 2, sr, evt_wgt);
+        if (cscShower->isTightInTime())
+          cscShowerOccupancyTight->Fill(chamber * 2, sr, evt_wgt);
+        if (cscShower->isLooseInTime())
+          cscShowerOccupancyLoose->Fill(chamber * 2 - 1, sr, evt_wgt);
+        if (cscShower->isNominalInTime())
+          cscShowerOccupancyNom->Fill(chamber * 2 - 1, sr, evt_wgt);
+        if (cscShower->isTightInTime())
+          cscShowerOccupancyTight->Fill(chamber * 2 - 1, sr, evt_wgt);
+      } else {
+        if (cscShower->isLooseInTime())
+          cscShowerOccupancyLoose->Fill(chamber, sr);
+        if (cscShower->isNominalInTime())
+          cscShowerOccupancyNom->Fill(chamber, sr);
+        if (cscShower->isTightInTime())
+          cscShowerOccupancyTight->Fill(chamber, sr);
+      }
+
+      cscShowerStationRing->Fill(0.5, sr);
+      if (cscShower->isLooseInTime())
+        cscShowerStationRing->Fill(1.5, sr);
+      if (cscShower->isNominalInTime())
+        cscShowerStationRing->Fill(2.5, sr);
+      if (cscShower->isTightInTime())
+        cscShowerStationRing->Fill(3.5, sr);
+
+      if (station > 1 && (ring % 2) == 1) {
+        cscShowerChamber->Fill(chamber * 2, 0.5, evt_wgt);
+        cscShowerChamber->Fill(chamber * 2 - 1, 0.5, evt_wgt);
+        if (cscShower->isLooseInTime())
+          cscShowerChamber->Fill(chamber * 2, 1.5, evt_wgt);
+        if (cscShower->isNominalInTime())
+          cscShowerChamber->Fill(chamber * 2, 2.5, evt_wgt);
+        if (cscShower->isTightInTime())
+          cscShowerChamber->Fill(chamber * 2, 3.5, evt_wgt);
+        if (cscShower->isLooseInTime())
+          cscShowerChamber->Fill(chamber * 2 - 1, 1.5, evt_wgt);
+        if (cscShower->isNominalInTime())
+          cscShowerChamber->Fill(chamber * 2 - 1, 2.5, evt_wgt);
+        if (cscShower->isTightInTime())
+          cscShowerChamber->Fill(chamber * 2 - 1, 3.5, evt_wgt);
+      } else {
+        cscShowerChamber->Fill(chamber, 0.5);
+        if (cscShower->isLooseInTime())
+          cscShowerChamber->Fill(chamber, 1.5);
+        if (cscShower->isNominalInTime())
+          cscShowerChamber->Fill(chamber, 2.5);
+        if (cscShower->isTightInTime())
+          cscShowerChamber->Fill(chamber, 3.5);
+      }
+    }
+  }
+
+  // Fill EMTF regional shower plots
+  for (auto Shower = EmtfShowers->begin(); Shower != EmtfShowers->end(); ++Shower) {
```

Comment:
```suggestion
  for (auto const& shower : EmtfShowers) {
```
(lowercase "s" is more in line with the rest of the code)